universal-application-tool-0.0.1/app/views/admin/programs/ProgramBlockEditView.java

```diff
@@ -135,44 +151,65 @@ public ContainerTag blockOrderPanel(ProgramDefinition program, long focusedBlock
                 Styles.W_1_5,
                 Styles.BORDER_R,
                 Styles.BORDER_GRAY_200);
+    addBlocksToContainer(
+        ret, program, program.getBlockDefinitions(Optional.empty()), focusedBlockId, 1);
+    ret.with(
+        submitButton("Add Block")
+            .withId("add-block-button")
+            .attr(Attr.FORM, CREATE_BLOCK_FORM_ID)
+            .withClasses(Styles.M_4));
+    return ret;
+  }
 
-    for (BlockDefinition block : program.blockDefinitions()) {
+  private void addBlocksToContainer(
+      ContainerTag container,
+      ProgramDefinition programDefinition,
+      ImmutableList<BlockDefinition> blockDefinitions,
+      long focusedBlockId,
+      int level) {
+    for (BlockDefinition blockDefinition : blockDefinitions) {
       String editBlockLink =
-          controllers.admin.routes.AdminProgramBlocksController.edit(program.id(), block.id())
+          controllers.admin.routes.AdminProgramBlocksController.edit(
+                  programDefinition.id(), blockDefinition.id())
               .url();
 
       // TODO: Not i18n safe.
-      int numQuestions = block.getQuestionCount();
-      String questionCountText = String.format("%d Question", numQuestions);
-      if (numQuestions != 1) {
-        questionCountText += 's';
-      }
-      String blockName = block.name();
+      int numQuestions = blockDefinition.getQuestionCount();
+      String questionCountText = String.format("Question count: %d", numQuestions);
+      String blockName = blockDefinition.name();
 
       ContainerTag blockTag =
           a().withHref(editBlockLink)
               .with(p(blockName), p(questionCountText).withClasses(Styles.TEXT_SM));
-      String selectedClasses = block.id() == focusedBlockId ? Styles.BG_GRAY_100 : "";
+      String selectedClasses = blockDefinition.id() == focusedBlockId ? Styles.BG_GRAY_100 : "";
       blockTag.withClasses(Styles.BLOCK, Styles.PY_2, Styles.PX_4, selectedClasses);
-      ret.with(blockTag);
-    }
 
-    ret.with(
-        submitButton("Add Block")
-            .withId("add-block-button")
-            .attr(Attr.FORM, CREATE_BLOCK_FORM_ID)
-            .withClasses(Styles.M_4));
-    return ret;
+      container.with(blockTag);
+
+      // Recursively do this
```

Comment:
Suggestion (blocking): "do this" is vague, please elaborate

Reply:
fixed